tools/cldr-code/src/main/java/org/unicode/cldr/util/personname/PersonNameFormatter.java

```diff
@@ -1297,6 +1309,31 @@ public PersonNameFormatter(CLDRFile cldrFile) {
         this.fallbackFormatter = new FallbackFormatter(new ULocale(cldrFile.getLocaleID()), initialPattern, initialSequencePattern);
     }
 
+    /**
+     * Utility for constructing data from path and value.
+     */
+    public static Pair<ParameterMatcher, NamePattern> fromPathValue(XPathParts parts, String value) {
+
+        //ldml/personNames/personName[@length="long"][@usage="sorting"]/namePattern[alt="2"]
+        // value = {surname}, {given} {given2} {suffix}
+        final String altValue = parts.getAttributeValue(-1, "alt");
+        int rank = altValue == null ? 0 : Integer.parseInt(altValue);
+        ParameterMatcher pm = new ParameterMatcher(
+            hackFix(parts.getAttributeValue(-2, "length")),
+            parts.getAttributeValue(-2, "style"),
+            parts.getAttributeValue(-2, "usage"),
+            parts.getAttributeValue(-2, "order")
+            );
+
+        // TODO change en.xml to not have periods after initials
```

Comment:
en.xml was changed in #1869 so maybe remove the next line?

Reply:
got it, thanks